### ISSUE TITLE
Annotate C# scripts with XML comments

### ISSUE THREAD

**Issue by AnonymousFool** [Micky774/unity_project#15]
I took the opportunity in enemy_abcs, the branch associated with #7 , to annotate the code of EnemyBehaviour.cs with [XML comments](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/xmldoc/recommended-tags) describing variables, functions, and classes because I read that it's useful in C# codebases. You can pull that branch to play around with the file yourself. Here I'm going to explain what I find to be the two core benefits I've discovered using screenshots based on the content of this branch.

Note that screenshots depicted are a proof of concept on code and comments that have been reviewed by no-one so far, and thus are very subject to change.

1. Intellisense displays your descriptive comments when appropriate. When you hover over a variable, function, etc. you're using, or when intellisense gives that member as a recommendation, you receive the relevant descriptive information from the XML comments in the intellisense popup. This would help us keep our code organized in the same way that our naming conventions do: we can gain information on variables and functions at a glance without having to dig back through older code.
![intellisenseXML](https://user-images.githubusercontent.com/34871200/236480070-4bc99db8-d7fc-46ab-9de3-a0f858fcc0cd.png)

2. We can use [doxygen](https://www.doxygen.nl/) on our source codebase with XML-commented code to automatically generate clean documentation .html files. We could use a hosting service like github pages to have an online wiki for ourselves of the code we've written, which will become more and more useful as our codebase expands, or we could just generate and store the .html files locally. The only additional thing we'd need to store in the repository is an ~110 KB Doxyfile providing the documentation generation settings.
![doxygenDocumentation](https://user-images.githubusercontent.com/34871200/236480095-310f6fc2-5f6e-42a7-b787-e13adbb0822c.png)

Apparently, it's also capable of generating a pdf manual, but I haven't managed to successfully use this feature yet. Feel free to play around with doxygen yourself locally.

Given these features, I'm going to suggest broader use of XML comments throughout new code, as well as a task to update old code with XML comments. Discuss.

**Comment by Micky774** [comment]
I think it is reasonable to enforce these for public methods / variables, and leave it optional for private. Overall it does add a great deal of value, especially as we grow out and each of us take on larger scope projects and feature work.

**Comment by AnonymousFool** [comment]
Alright, I've updated the MUGA with the XML comment standard. From now on, when we add a public class or public member to a class in a branch, we should annotate it with XML comments.

Currently #5 adds XML comments for the contents of the main branch, and since I used the enemy_abcs branch as an experiment, I've added XML comments to the new classes in #7 as well.

I think for now I can close this issue. I've added a task to Notion to add XML comments to #18 , and though it's obviously not complete, specific implementations of XML comments were not the reason I made this issue, it was for discussion of whether we wanted XML comments in the first place. A discussion we've now had.
